### PR TITLE
Display range chart with BH/H BCE/CE labels

### LINKED ISSUE #895: Update how date range slider displays Hijri dates
I'm creating a new ticket for this to clearly separate this issue from:

- General styling/display issues with the date range slider (covered in #765)
- Display of Hijri dates in metadata (covered in #845)

As noted in #845, the concern is that (if I understand correctly) the Hijri calendar starts with year 1, so displaying a value like "-500" as a label on the x-axis of the date range graph and as a value in the date range inputs doesn't really make sense.

### Preferred solution
It's not clear to @jacobthill and I what is technically feasible but here is our preferred solution:

1. For negative year values displayed on the x-axis, drop the minus sign and append "BH" to the year
2. For positive year values displayed on the x-axis, append "H" to the year
3. Do the same as 1 and 2 above for years displayed in the input fields
4. Do the same as 1 and 2 above for years displayed in the active constraint label

71283802-381bc300-232f-11ea-918a-59635810489e

(Ignore the flash message shown next to the facet title; this would be very challenging to display nicely in the sidebar version of the facet, and I'm hoping it is not really necessary.)

 ### Potential lesser solution
This is not at all ideal, but I'm worried that the input fields (item 3 above) might be more challenging than the other items, because it needs to be strictly numeric. If that's true, we might be able to get away with just doing:

1. For negative year values displayed on the x-axis, drop the minus sign and append "BH" to the year
2. For positive year values displayed on the x-axis, append "H" to the year
3. Do the same as 1 and 2 above for years displayed in the active constraint label

and leaving the input fields displaying a minus sign, to keep the field numeric-only. It would be very awkward to use different formatting for the input fields versus the labels, but perhaps it will nonetheless be clear to the user (because the numeric values will still match) that "-500" in the input box is shown as "500 BH" on the x-axis. 

I really don't like this solution, but bring it up as possibly better than doing nothing at all, if we can't do all of the preferred solution.

### PR DESCRIPTION
**Pull request by mejackreed**
Fixes #895 
Adds BCE/CE and BH/H suffixes to range limit and constraints in both English and Arabic. Confirmed correct behavior on consultation with @ggeisler and @jacobthill .

*Note* the behavior in the screenshot represents integration with https://github.com/projectblacklight/blacklight_range_limit/pull/148. This needs to be released and shipped (or similar) for everything to work properly.

## Updated screen shots
Screen Shot 2020-04-07 at 13 38 01
Screen Shot 2020-04-07 at 13 37 17

## Old Screenshot
Screen Shot 2020-04-03 at 9 58 52 AM

### CODE REVIEW

**Comment by ggeisler** [comment]
@mejackreed Great to see we're able to add the date abbreviations to the numeric value. I realize that this is really challenging our horizontal space but I think we might need to include a space between the date's numeric value and the "BH" or "H" suffix. 

Can we try that and see how it looks?

**Comment by mejackreed** [comment]
Sure thing. Here is an updated shot with the spaces.
Screen Shot 2020-04-03 at 10 20 35 AM

**Comment by ggeisler** [comment]
👍 Thanks, I think we should go with the version with the space.

File: app/assets/javascripts/range_limit_config.js [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+  });
+  
+  var gregorianConfig = $.extend({}, config, {
+    xaxis: {
+      tickFormatter: genericTickFormatter(['bce', 'ce'])
+    }
+  })
+
+  $('.blacklight-cho_date_range_norm_isim').data('plot-config', gregorianConfig);
+  $('.blacklight-cho_date_range_hijri_isim').data('plot-config', hijriConfig);
```
Comment (mejackreed) [inline app/assets/javascripts/range_limit_config.js:32]: Once we get a new release of BLRL, we can extend this to also do this on modal open.

File: app/helpers/override_range_limit_helper.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Override range limit provided helpers
+module OverrideRangeLimitHelper
+  include RangeLimitHelper
+
+  ##
+  # Overriden helper to add the BH/H and BCE/CE to labels
+  def range_display(solr_field, myparams = params)
+    display = super(solr_field, myparams)
```
Comment (jkeck) [inline app/helpers/override_range_limit_helper.rb:10]: I _believe_ `super` w/o args passes through the args methods to the super method.

Comment (mejackreed) [reply app/helpers/override_range_limit_helper.rb]: Oh good call. I've updated accordingly.

**Pull request merged**